decode_morse.rb

```diff
@@ -0,0 +1,37 @@
+def decode_char(data)
+  alphabet = { '.-' => 'A', '-...' => 'B', '-.-.' => 'C',
+               '-..' => 'D', '.' => 'E', '..-.' => 'F', '--.' => 'G',
+               '....' => 'H', '..' => 'I', '.---' => 'J', '-.-' => 'K',
+               '.-..' => 'L', '--' => 'M', '-.' => 'N', '---' => 'O',
+               '.--.' => 'P', '--.-' => 'Q', '.-.' => 'R', '...' => 'S',
+               '-' => 'T', '..-' => 'U', '...-' => 'V', '.--' => 'W',
+               '-..-' => 'X', '-.--' => 'Y', '--..' => 'Z', '.----' => '1',
+               '..---' => '2', '...--' => '3', '....-' => '4', '.....' => '5',
+               '-....' => '6', '--...' => '7', '---..' => '8', '----.' => '9',
+               '-----' => '0' }
+  alphabet[data]
+end
+
+def decode_word(word)
+  decode = ''
+  character = ''
+  spaces = 0
+  word.each_char do |ch|
+    if ch == ' '
+      spaces += 1
+      if character.length.positive?
+        decode += decode_char character
+        character = ''
+      end
+    else
+      character += ch
+      decode += ' ' if spaces > 1
+      spaces = 0
+    end
+  end
+  decode += decode_char character if character.length.positive?
+  decode
+end
+
+print "#{decode_word('-- -.--   -. .- -- .')}\n"
+print decode_word('.-   -... --- -..-   ..-. ..- .-.. .-..   --- ..-.   .-. ..- -... .. . ...')
```

Comment:
- [ ]  You're expected to create a method `decode` and should have these features:

- it splits input sentences by 3 spaces (word delimiter in Morse code)
- it iterates through all words from the input sentence
- it reuses the `#decode_word` method to translate a single encoded-word into a word (DRY rule)
- it combines translated words into a whole sentence to be returned as the final output

### SAPLE WAY TO USE IT:
`p decode("-- -.-- -. .- -- .")` or `decode("-- -.-- -. .- -- .")` should return `"MY NAME"`.`

### Instructions:
![image](https://github.com/alvaroiporre/Decode-Morse-Code/assets/50990925/5ed3a4a6-db13-4e54-80f6-b269d72fec50)